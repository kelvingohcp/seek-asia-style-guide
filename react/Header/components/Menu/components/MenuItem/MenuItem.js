import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { ListItem } from 'seek-asia-style-guide/react';
import styles from './MenuItem.less';

const interactionButton = ({ ItemIcon, children, linkUrl, linkRenderer, ...restProps }) => {
  return (
    linkRenderer({
      href: linkUrl !== '' || typeof(linkUrl) !== 'undefined' ? linkUrl : '',
      className: styles.menuText,
      children: (
        <ListItem
          value={children}
          icon={ItemIcon}
          hasHoverState
          {...restProps}
        />
      )
    })
  );
};

interactionButton.propTypes = {
  ItemIcon: PropTypes.func,
  children: PropTypes.any,
  linkUrl: PropTypes.string,
  linkRenderer: PropTypes.func
};

const MenuItem = ({ linkUrl, handleClick, children, ItemIcon, linkRenderer, className, title, ...restProps }) => {
  const interactionProps = {
    className: classnames(styles.root, className),
    children,
    ItemIcon,
    linkUrl,
    title,
    linkRenderer
  };

  if (linkUrl) {
    return interactionButton({
      ...interactionProps
    });
  } else if (handleClick) {
    return interactionButton({
      ...interactionProps,
      onClick: handleClick
    });
  }

  return (
    <ListItem
      className={styles.root}
      value={children}
      icon={ItemIcon}
      {...restProps}
    />
  );
};

MenuItem.propTypes = {
  linkUrl: PropTypes.string,
  handleClick: PropTypes.func,
  children: PropTypes.any,
  ItemIcon: PropTypes.element,
  linkRenderer: PropTypes.func,
  className: PropTypes.string,
  title: PropTypes.string
};

export default MenuItem;
