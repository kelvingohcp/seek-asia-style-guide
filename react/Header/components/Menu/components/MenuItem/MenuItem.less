@import (reference) "~seek-asia-style-guide/theme";

.root {
  min-height: @grid-base * 12;
  background-color: @saWhite;
  border-bottom: 1px solid @saGrey5;
}

.item {
  display: flex;
  align-items: center;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: flex-start;
  width: 100%;
  height: 100%;
  padding: 0 @grid-base * 4;
  color: @saGrey1;
}

.item:hover,
.item:active {
  text-decoration: none;
}

.item:active {
  filter: opacity(50%);
}

.item:focus {
  outline: none;
}

.noIcon {
  height: @grid-base * 12;
}

.icon {
  padding-top: @grid-base * 2;
  margin-right: @grid-base * 4;
}

/* stylelint-disable */
/* we need to avoid specificity problems from unpredictable order of imports between icons and MenuItem */
.icon svg {
  width: @grid-base * 5 !important;
  height: @grid-base * 5 !important;
}
/* stylelint-enable */

.menuText {
  text-decoration: none;
}
