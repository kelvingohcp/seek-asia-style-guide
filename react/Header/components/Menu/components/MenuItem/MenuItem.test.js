import React from 'react';
import { shallow } from 'enzyme';

import MenuItem from './MenuItem';

describe('MenuItem', () => {
  const FakeIcon = () => {
    return (
      <div>I'm an icon!</div>
    );
  };

  const defaultProps = {
    brandStyles: {}
  };

  it('should render with no icon and default props', () => {
    const wrapper = shallow(<MenuItem {...defaultProps}>Awesome Brand Portal!</MenuItem>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with icon and default props', () => {
    const wrapper = shallow(<MenuItem ItemIcon={FakeIcon} {...defaultProps}>Awesome Brand Portal!</MenuItem>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with a linkUrl', () => {
    const testProps = {
      ...defaultProps,
      linkUrl: 'https://brand.seekasia.com/'
    };
    const wrapper = shallow(<MenuItem {...testProps}>Awesome Brand Portal!</MenuItem>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with a click handler', () => {
    const testProps = {
      ...defaultProps,
      handleClick: jest.fn()
    };
    const wrapper = shallow(<MenuItem {...testProps}>Awesome Brand Portal!</MenuItem>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with all optional props', () => {
    const testProps = {
      ...defaultProps,
      brandStyles: {
        menuIcon: 'menuIcon'
      },
      iconProps: {
        direction: 'left',
        svgClassName: 'awesomeSVG'
      },
      className: 'awesomeClass',
      itemClass: 'seriously I thought css modules would solve this precarious css specificity nonsense'
    };
    const wrapper = shallow(<MenuItem ItemIcon={FakeIcon} {...testProps}>Awesome Brand Portal!</MenuItem>);
    expect(wrapper).toMatchSnapshot();
  });
});
