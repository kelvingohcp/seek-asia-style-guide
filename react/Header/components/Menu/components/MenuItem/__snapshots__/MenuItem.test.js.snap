// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem should render with a click handler 1`] = `""`;

exports[`MenuItem should render with a linkRenderer 1`] = `
<ListItem
  activeState={false}
  badge={null}
  brandStyles={Object {}}
  checkbox={false}
  className="root"
  compact={false}
  disableBackground={false}
  hasHoverState={false}
  icon={null}
  noShadow={false}
  secondLevel={false}
  value="Awesome Brand Portal!"
/>
`;

exports[`MenuItem should render with all optional props 1`] = `
<ListItem
  activeState={false}
  badge={null}
  brandStyles={
    Object {
      "menuIcon": "menuIcon",
    }
  }
  checkbox={false}
  className="root"
  compact={false}
  disableBackground={false}
  hasHoverState={false}
  icon={[Function]}
  iconProps={
    Object {
      "direction": "left",
      "svgClassName": "awesomeSVG",
    }
  }
  itemClass="seriously I thought css modules would solve this precarious css specificity nonsense"
  noShadow={false}
  secondLevel={false}
  value="Awesome Brand Portal!"
/>
`;

exports[`MenuItem should render with icon and default props 1`] = `
<ListItem
  activeState={false}
  badge={null}
  brandStyles={Object {}}
  checkbox={false}
  className="root"
  compact={false}
  disableBackground={false}
  hasHoverState={false}
  icon={[Function]}
  noShadow={false}
  secondLevel={false}
  value="Awesome Brand Portal!"
/>
`;

exports[`MenuItem should render with no icon and default props 1`] = `
<ListItem
  activeState={false}
  badge={null}
  brandStyles={Object {}}
  checkbox={false}
  className="root"
  compact={false}
  disableBackground={false}
  hasHoverState={false}
  icon={null}
  noShadow={false}
  secondLevel={false}
  value="Awesome Brand Portal!"
/>
`;
